### ISSUE TITLE
Diff Test, expected collisions do not affect the testing result

### ISSUE THREAD

**Issue by gzm55** [rurban/smhasher#199]
In `Diff` Test, we have an excepted collision count:
https://github.com/rurban/smhasher/blob/d428f2f6a427e86b41b1bbbcdefc3e03aff28576/DifferentialTest.h#L141

But this variable seems do not affect the testing result:
https://github.com/rurban/smhasher/blob/d428f2f6a427e86b41b1bbbcdefc3e03aff28576/DifferentialTest.h#L44

and the last group collisions are also skipped for unsetting the `result`.
https://github.com/rurban/smhasher/blob/d428f2f6a427e86b41b1bbbcdefc3e03aff28576/DifferentialTest.h#L68

**Comment by rurban** [comment]
Yes, the calculation of `expected` in line 141 is only for the print before we actually check the collisions.

We don't check against the float calc of expected,  we check only against 3 or more collisions as described in the comment.
_Sort through the differentials, ignoring collisions that only occured once (these could be false positives). If we find collisions of 3 or more, the differential test fails._

A much simpler test.

**Comment by gzm55** [comment]
ok, but the last group should be counted, in the block including line 68.
